blocklytest/src/androidTest/java/com/google/blockly/model/WorkspaceTest.java

```diff
@@ -94,6 +95,31 @@ public void testSerialization() throws BlocklySerializerException {
         assertThat(os.toString()).isEqualTo(EMPTY_WORKSPACE);
     }
 
+    @Test
+    public void testWorkspaceIdSetOnNewRootBlocks() {
+        // given
+        Block mockBlock = Mockito.mock(Block.class);
+
+        // when
+        mWorkspace.addRootBlock(mockBlock, true);
+
+        // then
+        Mockito.verify(mockBlock, Mockito.times(1)).setEventWorkspaceId(mWorkspace.getId());
+    }
+
+    @Test
+    public void testWorkspaceIdUnsetOnRootBlockRemoved() {
+        // given
+        Block mockBlock = Mockito.mock(Block.class);
+        mWorkspace.addRootBlock(mockBlock, true);
+
+        // when
+        mWorkspace.removeRootBlock(mockBlock, false);
+
+        // then
+        Mockito.verify(mockBlock, Mockito.times(1)).setEventWorkspaceId(null);
+    }
+
     private static ByteArrayInputStream assembleWorkspace(String interior) {
```

Comment:
I'm also looking to test the workspace id on the related BlocklyController methods: addRootBlock(), trashRootBlock(), addBlockFromTrash(), trashAllBlocks(), resetWorkspace(), removeBlockTree().